hyperspy/models/edsmodel.py

```diff
@@ -0,0 +1,822 @@
+# -*- coding: utf-8 -*-
+# Copyright 2007-2011 The HyperSpy developers
+#
+# This file is part of  HyperSpy.
+#
+#  HyperSpy is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+#  HyperSpy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with  HyperSpy.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import division
+
+import numpy as np
+import math
+
+from hyperspy.models.model1D import Model1D
+from hyperspy._signals.eds import EDSSpectrum
+from hyperspy.misc.elements import elements as elements_db
+from hyperspy.misc.eds import utils as utils_eds
+import hyperspy.components as create_component
+
+
+def _get_weight(element, line, weight_line=None):
+    if weight_line is None:
+        weight_line = elements_db[
+            element]['Atomic_properties']['Xray_lines'][line]['weight']
+    return lambda x: x * weight_line
+
+
+def _get_iweight(element, line, weight_line=None):
+    if weight_line is None:
+        weight_line = elements_db[
+            element]['Atomic_properties']['Xray_lines'][line]['weight']
+    return lambda x: x / weight_line
+
+
+def _get_sigma(E, E_ref, units_factor):
+    # 2.5 from Goldstein, / 1000 eV->keV, / 2.355^2 for FWHM -> sigma
+    return lambda sig_ref: math.sqrt(abs(
+        4.5077 * 1e-4 * (E - E_ref) * units_factor + np.power(sig_ref, 2)))
```

Comment:
Magic numbers are Bad, but at least it is documented in the comment somehow (Goldstein? What is order of division? Is eV->keV being handled correctly together with `units_factor`? This can definitely be made more readable / commented more).